src/hotspot/share/runtime/continuationEntry.cpp

```diff
@@ -111,9 +111,11 @@ void ContinuationEntry::describe(FrameValues& values, int frame_no) const {
 #endif
 
 #ifdef ASSERT
-bool ContinuationEntry::assert_entry_frame_laid_out(JavaThread* thread) {
+bool ContinuationEntry::assert_entry_frame_laid_out(JavaThread* thread, bool preempted) {
   assert(thread->has_last_Java_frame(), "Wrong place to use this assertion");
 
+  if (preempted) return true;
```

Comment:
I don't see the point of adding a new parameter just so the method can check it and return immediately. Shouldn't the caller be checking this?